### PR TITLE
Making file_server reusable

### PR DESCRIPTION
**Pull request by avinashdvv**
making the file_server file a function for reusability

cc: @kevinkassimo

### CODE REVIEW

File: net/http.ts [hunk 1]
```diff
@@ -323,3 +324,13 @@ async function readAllIterator(
...
 }
+
+const http = {
+  Handle: function() {
+
+  },
+  FileServer,
+};
+
+export default http;
```
Comment (zhmushan) [inline net/http.ts]: I think users can import directly from `file_serve.ts` instead of `http.ts`.

File: net/http.ts [hunk 2]
```diff
@@ -323,3 +324,13 @@ async function readAllIterator(
   }
   return collected;
 }
+
+const http = {
+  Handle: function() {
+
```
Comment (kevinkassimo) [inline net/http.ts]: What's this?

File: net/http.ts [hunk 3]
```diff
@@ -323,3 +324,13 @@ async function readAllIterator(
...
+  },
+  FileServer,
```
Comment (kevinkassimo) [inline net/http.ts]: I prefer naming this `fileServer`. Also, it feels more natural to me to actually move all the fileServer code to `http.ts`.

`file_server.ts` would look similar to somthing like this after the move:
```ts
import { cwd, args } from "deno";
import { fileServer } from "./http.ts";

const serverArgs = args.slice();
let CORSEnabled = false;
for (let i = 0; i < serverArgs.length; i++) {
  if (serverArgs[i] === "--cors") {
    CORSEnabled = true;
    serverArgs.splice(i, 1);
    break;
  }
}
let currentDir = cwd();
const target = serverArgs[1];
if (target) {
  currentDir = `${currentDir}/${target}`;
}
const addr = `0.0.0.0:${serverArgs[2] || 4500}`;

fileServer(currentDir, addr, {
  cors: CORSEnabled,
})
```

File: net/http.ts (hunk 1 above)
Comment (bartlomieju) [reply net/http.ts]: AFAICT we prefer named exports over default ones.

Comment (ry) [reply net/http.ts]: Yes - let’s avoid default exports.

File: net/http.ts [hunk 4]
```diff
@@ -323,3 +353,185 @@ async function readAllIterator(
...
+    res.headers = new Headers();
+  }
+  res.headers!.append("access-control-allow-origin", "*");
+  res.headers!.append(
+    "access-control-allow-headers",
+    "Origin, X-Requested-With, Content-Type, Accept, Range"
+  );
+}
+
+export function fileServer(currentDir, addr, { cors }) {
```
Comment (bartlomieju) [inline net/http.ts]: @avinashdvv please add types to arguments

File: net/file_server.ts [hunk 5]
```diff
@@ -58,202 +27,7 @@ if (target) {
...
-
-  return extensionsMap[''];
-}
-
-async function serveFile(req: ServerRequest, filename: string) {
-  const file = await open(filename);
-  const fileInfo = await stat(filename);
-  const headers = new Headers();
-  headers.set("content-length", fileInfo.len.toString());
-  headers.set("content-type", guessContentType(filename));
```
Comment (bartlomieju) [inline net/file_server.ts]: This bit is missing in new version and causes tests to fail.

**Comment by bartlomieju** [comment]
I'm not sure if moving most of the logic of file server to `http.ts` is a good idea. IMHO optimally we should leave code as is in `file_server.ts` but figure out a way to run server if that file is run directly, kinda like Python's:
```
def fileServer(...):
   ...

if __name__ == '__main__':
    fileServer(...)
```

**Comment by kevinkassimo** [comment]
@bartlomieju previous dicussion: 
https://github.com/denoland/deno_std/pull/15#issuecomment-446380595

**Pull request closed without merging**